mm.c

```diff
@@ -366,6 +417,111 @@ static void test_MAIN_FREE_LIST_INIT()
     printf("Test passed.\n\n");
 }
 
+static void test_PREV_NEXT_IN_HEAP() {
+	printf("Testing PREV_NEXT_IN_HEAP.\n");
+
+	size_t prev_size = 4;
+	size_t curr_size = 10;
+	size_t total_needed_test_size = prev_size + curr_size + HDR_FTR_SIZE*2;
+
+	char **ptr = malloc(WORD_SIZE * total_needed_test_size);
+	char **p_prev = ptr;
+	char **p_curr = ptr + HDR_FTR_SIZE + prev_size;
+
+	PUT_WORD(p_prev, PACK(prev_size, FREE));
+	PUT_WORD(FTRP(p_prev), PACK(prev_size, FREE));
+
+	PUT_WORD(p_curr, PACK(curr_size, FREE));
+	PUT_WORD(FTRP(p_curr), PACK(curr_size, FREE));
+
+	char **tmp = ((char **)(p_curr) - GET_SIZE((char **)(p_curr) - 1) - HDR_FTR_SIZE);
+
+	assert(p_prev == PREV_BLOCK_IN_HEAP(p_curr));
+	assert(p_curr == NEXT_BLOCK_IN_HEAP(p_prev));
+
+	printf("Test passed.\n\n");
+}
+
+static void test_coalesce() {
```

Comment:
Quan's writing the two functions place_block_into_free_list and remove_block_from_free_list